File: DevTask/app/src/main/java/dev/espera/devtask/domain/models/Task.kt
package dev.espera.devtask.domain.models

import dev.espera.devtask.common.domain.model.TaskPriority
import dev.espera.devtask.common.domain.model.TaskStatus
import java.time.LocalDateTime

data class Task(
    val id: String,
    val order: Int,
    val title: String,
    val tags: List<Tag>,
    val status: TaskStatus,
    val description: String?,
    val assignees: List<User>,
    val priority: TaskPriority,
    val dueDate: LocalDateTime,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
)

File: frontend/lib/types/task.ts
export interface Task {
  id: string;
  title: string;
  description: string | null;
  status: string;
  order: number;
  dueDate: string | null;
  priority: string;
  createdAt: string;
  updatedAt: string;
  userId: string;
  parentId: string | null;
  subtasks?: Task[];
  tags: Tag[] | [];
  user?: User;
  assignees: User[];
  parent?: Task | null;
}

export interface Tag {
  id: string;
  name: string;
  color: string | null;
  createdAt: string;
  updatedAt: string;
}

export interface User {
  id: string;
  email: string;
  firstName: string | null;
  lastName: string | null;
  createdAt: string;
  updatedAt: string;
}

export enum TaskStatus {
  TODO = "TODO",
  IN_PROGRESS = "IN_PROGRESS",
  DONE = "DONE",
}

export enum TaskPriority {
  LOW = "LOW",
  MEDIUM = "MEDIUM",
  HIGH = "HIGH",
}
